lib/new_relic/agent/instrumentation/stripe.rb

```diff
@@ -0,0 +1,28 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/instrumentation/stripe_subscriber'
+
+DependencyDetection.defer do
+  named :stripe
+
+  depends_on do
+    NewRelic::Agent.config[:'instrumentation.stripe'] == 'enabled'
+  end
+
+  depends_on do
+    defined?(Stripe) &&
+      Gem::Version.new(Stripe::VERSION) >= Gem::Version.new('5.15.0') # Stripe subscribers added 5.15.0
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing Stripe instrumentation')
+  end
+
+  executes do
```

Comment:
I think this block of code needs a `rescue` with an error level log message, so as to not crash the agent if either `subscribe` call fails. Optionally, a new method can be defined in `stripe_subscriber.rb` that handles the rescuing and that method could be called with a one-liner here.

Reply:
Yup yup yup this is important--good catch https://github.com/newrelic/newrelic-ruby-agent/pull/2180/commits/8cd8f196a25cb78da3da8b2130fb11b3d9202cb2

Reply:
Ooo on second look, the `executes` method already has a rescue. So I think we are covered here. Reverted. https://github.com/newrelic/newrelic-ruby-agent/pull/2180/commits/89b9db0edf085ab245b05b21e78b9610639f7e99